llvm/include/llvm/CodeGen/MachinePassManager.h

```diff
@@ -170,6 +192,17 @@ class MachineFunctionPassManager
     addRunOnModule<PassT>(P);
   }
 
+  // Avoid diamond problem.
```

Comment:
can you clarify that we'd be getting `PassInfoMixin` twice if we directly inherited from it (`PassManager` also inheriting from it)